osc-server/src/main/java/org/osc/core/broker/service/DAConformService.java

```diff
@@ -0,0 +1,198 @@
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.osc.core.broker.job.Job;
+import org.osc.core.broker.job.Job.JobCompletionListener;
+import org.osc.core.broker.job.JobEngine;
+import org.osc.core.broker.job.Task;
+import org.osc.core.broker.job.TaskGraph;
+import org.osc.core.broker.job.lock.LockObjectReference;
+import org.osc.core.broker.job.lock.LockObjectReference.ObjectType;
+import org.osc.core.broker.job.lock.LockRequest;
+import org.osc.core.broker.model.entities.appliance.DistributedAppliance;
+import org.osc.core.broker.model.entities.job.JobRecord;
+import org.osc.core.broker.model.entities.management.ApplianceManagerConnector;
+import org.osc.core.broker.model.entities.virtualization.openstack.DeploymentSpec;
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.service.api.ConformServiceApi;
+import org.osc.core.broker.service.exceptions.VmidcBrokerValidationException;
+import org.osc.core.broker.service.persistence.DeploymentSpecEntityMgr;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.request.ConformRequest;
+import org.osc.core.broker.service.response.BaseJobResponse;
+import org.osc.core.broker.service.tasks.conformance.DAConformanceCheckMetaTask;
+import org.osc.core.broker.service.tasks.conformance.DowngradeLockObjectTask;
+import org.osc.core.broker.service.tasks.conformance.UnlockObjectMetaTask;
+import org.osc.core.broker.service.tasks.conformance.UnlockObjectTask;
+import org.osc.core.broker.service.tasks.conformance.manager.MCConformanceCheckMetaTask;
+import org.osc.core.broker.service.tasks.conformance.securitygroupinterface.MgrSecurityGroupInterfacesCheckMetaTask;
+import org.osc.core.broker.service.transactions.CompleteJobTransaction;
+import org.osc.core.broker.service.transactions.CompleteJobTransactionInput;
+import org.osc.core.common.job.TaskGuard;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = {ConformServiceApi.class, DAConformService.class})
```

Comment:
should only expose DAConformService. ConformServiceApi should be deleted as the corresponding implementation is deleted as well

create a DAConformServiceApi, DSConformServiceAPI etc

Reply:
The DAConformService is the only one used outside the osc-server package, so adding the Api interfaces for the others will not decouple them for anyone who doesn't already know about the implementation. 

I could simply rename ConformServiceApi to DAConformServiceApi and forget about the other four. Adding all these methods that start jobs into the Api is not possible, since then osc-service-api would have to know about the Job class.